runtime/shared/shrclssup.c

```diff
@@ -324,7 +346,31 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void* reserved)
 				}
 				return J9VMDLLMAIN_FAILED;
 			}
+		} else if (shareOrphansArgIndex2 < shareOrphansArgIndex1) {
+			/* -Xshareclasses is not specified, but -XX:+ShareOrphans is specified. */
+			vm->sharedCacheAPI->sharedCacheEnabled = TRUE;
+			/* -XX:+ShareOrphans automatically turns on -Xshareclasses. Set xShareClassesPresent to TRUE */
+			vm->sharedCacheAPI->xShareClassesPresent = TRUE;
+			parseXXOptions(vm, &runtimeFlags);
+			runtimeFlags2 |= J9SHR_RUNTIMEFLAG2_ENABLE_CACHEORPHAN;
+			if (runtimeFlags & J9SHR_RUNTIMEFLAG_ENABLE_PERSISTENT_CACHE) {
```

Comment:
Pls add a helper function for setting the cache type, and another for the J9VM_OPT_JITSERVER section so don't have duplicated code and somebody could miss updating in the future.

Reply:
Added a new function.